packages/fxa-payments-server/src/lib/hooks.tsx

```diff
@@ -14,3 +18,23 @@ export function useCheckboxState(defaultState: boolean = false): useCheckboxStat
   const onChanged = useCallback(ev => setState(ev.target.checked), [ setState ]);
   return [ state, onChanged ];
 }
+
+export function useClickOutsideEffect<T>(onClickOutside: Function) {
```

Comment:
A comment here would be helpful. Reading up on refs and effects, I see that it's meant to be used in a certain way, but it involves a bunch of new patterns and a very specific usage pattern.

Reply:
Added a JSDoc-style comment here